liblouis/compileTranslationTable.c

```diff
@@ -2672,93 +2667,95 @@ compileRule(FileInfo *nested, TranslationTableHeader **table,
 	case CTO_LitDigit:
-		compileCharDef(nested, opcode, CTC_LitDigit, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_LitDigit, noback, nofor, table, displayTable);
 	case CTO_Punctuation:
-		compileCharDef(
+		return compileCharDef(
 				nested, opcode, CTC_Punctuation, noback, nofor, table, displayTable);
-		break;
 	case CTO_Math:
-		compileCharDef(nested, opcode, CTC_Math, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_Math, noback, nofor, table, displayTable);
 	case CTO_Sign:
-		compileCharDef(nested, opcode, CTC_Sign, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_Sign, noback, nofor, table, displayTable);
 	case CTO_Letter:
-		compileCharDef(nested, opcode, CTC_Letter, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_Letter, noback, nofor, table, displayTable);
 	case CTO_UpperCase:
-		compileCharDef(nested, opcode, CTC_UpperCase, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_UpperCase, noback, nofor, table, displayTable);
 	case CTO_LowerCase:
-		compileCharDef(nested, opcode, CTC_LowerCase, noback, nofor, table, displayTable);
-		break;
+		return compileCharDef(
+				nested, opcode, CTC_LowerCase, noback, nofor, table, displayTable);
 	case CTO_Grouping:
-		ok = compileGrouping(nested, noback, nofor, table, displayTable);
-		break;
+		return compileGrouping(nested, noback, nofor, table, displayTable);
 	case CTO_UpLow:
-		ok = compileUplow(nested, noback, nofor, table, displayTable);
-		break;
+		return compileUplow(nested, noback, nofor, table, displayTable);
 	case CTO_Display:
-		if (!displayTable) break;
-		if (getRuleCharsText(nested, &ruleChars))
-			if (getRuleDotsPattern(nested, &ruleDots)) {
-				if (ruleChars.length != 1 || ruleDots.length != 1) {
-					compileError(
-							nested, "Exactly one character and one cell are required.");
-					ok = 0;
+		if (displayTable) {
+			if (getRuleCharsText(nested, &ruleChars))
+				if (getRuleDotsPattern(nested, &ruleDots)) {
+					if (ruleChars.length != 1 || ruleDots.length != 1) {
+						compileError(nested,
+								"Exactly one character and one cell are required.");
+						return 0;
+					}
+					putCharDotsMapping(
+							nested, ruleChars.chars[0], ruleDots.chars[0], displayTable);
+					return 1;
 				}
-				putCharDotsMapping(
-						nested, ruleChars.chars[0], ruleDots.chars[0], displayTable);
-			}
-		break;
+			return 0;
+		}
+		return 1;
 	/* now only opcodes follow that don't modify the display table */
 	default:
-		if (!table) break;
+		if (!table) return 1;
 		switch (opcode) {
 		case CTO_None:
-			break;
+			return 0;
 		case CTO_Locale:
 			compileWarning(nested,
 					"The locale opcode is not implemented. Use the locale meta data "
 					"instead.");
-			break;
+			return 1;
```

Comment:
Presumably `return 1` implies "ignore this error"

Reply:
Yes, when it's a warning and not an error I ignore it.